### PR TITLE
Added code to allow template files to define imports they need.

### PR DESCRIPTION
**Pull request by intervalia**
Added code to allow template files to define imports they need.
Improved Docs.
Changed version to 1.1.0

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -133,12 +133,122 @@ If you have all of your components in the folder `./comps` then you would call `
 
 Every folder directly under `./comps` would get processed and the build tools and handed off to rollup.
```
Comment (GyozaGuy) [inline README.md]: There is something a bit off about this part of the sentence, but I'm not sure how to fix it: `and the build tools and handed off to rollup`

File: README.md [hunk 2]
```diff
@@ -133,12 +133,122 @@ If you have all of your components in the folder `./comps` then you would call `
...
 
+## Locale files
+
+Locale files are JSON files that contains locale specific translations of strings you use in your application. By default the files are all placed in the `locales` folder and are names `strings_??.json` where the `??` is the 2 letter locale such as `en`, `fr`, or `ja`.
```
Comment (GyozaGuy) [inline README.md]: `contains` should be `contain`
`names` should be `named`

File: README.md [hunk 3]
```diff
@@ -133,12 +133,122 @@ If you have all of your components in the folder `./comps` then you would call `
...
+
+The template file `person.html`:
+
+```html
+<div class="first">
+  <div>Name: <span class="name">Frank N Stein</span></div>
+</div>
+```
+
+Would before the template:
```
Comment (GyozaGuy) [inline README.md]: It seems like `before` was supposed to be `return` or `provide` or some other verb.

File: README.md [hunk 4]
```diff
@@ -133,12 +133,122 @@ If you have all of your components in the folder `./comps` then you would call `
...
+
+```JS
+  case 'person':
+    return `<div class="first"> <div>Name: <span class="name">Frank N Stein</span></div> </div>`;
+```
+
+### Using locale strings in templates
+
+If your component includes both templates and locale strings then the strings are made available to the template through the `lang` variable. The `lang` variable it auto-inserted into the `templates.mjs` file for all projects that have locale string files.
```
Comment (GyozaGuy) [inline README.md]: `The 'lang' variable it` should be `The 'lang' variable is`

**Review by GyozaGuy — APPROVED** [review]
Found some minor grammar issues in the readme, but otherwise looks good!

File: README.md (hunk 1 above)
Comment (intervalia) [reply README.md]: How about this:
> Every folder directly under `./comps` would get processed. Any subfolder that was found to match the requirements of a component will be handed off to rollup for further processing.

File: README.md (hunk 2 above)
Comment (intervalia) [reply README.md]: Fixed

File: README.md (hunk 3 above)
Comment (intervalia) [reply README.md]: Fixed

File: README.md (hunk 4 above)
Comment (intervalia) [reply README.md]: Fixed

File: README.md [hunk 5]
```diff
@@ -131,14 +131,124 @@ would still produce the same output structure of:
...
+
+### Using locale strings in templates
+
+If your component includes both templates and locale strings then the strings are made available to the template through the `lang` variable. The `lang` variable is auto-inserted into the `templates.mjs` file for all projects that have locale string files.
+
+Locale file: `./locales/strings_en.json`
+
+```JSON
+{
+  NAME: "Frank N Stein"
```
Comment (zvakanaka) [inline README.md]: This key should have double-quotes.

File: README.md [hunk 6]
```diff
@@ -131,14 +131,124 @@ would still produce the same output structure of:
...
+> If no locale key is provided in the `locales()` function then the default locale object is returned. 
+
+## Templates
+
+Template files tend to be HTML files but can be any kind of file. The templates are embedded into a template literal inside of the `template.mjs` file.
+
+The template file `person.html`:
+
+```html
+<div class="first">
```
Comment (zvakanaka) [inline README.md]: `first` is sort of confusing for a name example. Maybe `one` or something instead?

File: README.md (hunk 5 above)
Comment (intervalia) [reply README.md]: Good catch!

**Review by zvakanaka — APPROVED** [review]

**Review by zvakanaka — APPROVED** [review]

**Pull request merged**